shared/tests/utility/motion/TestSplines.cpp

```diff
@@ -280,7 +281,7 @@ TEST_CASE("Test Smooth Spline", "[utility][motion][splines][SmoothSpline]") {
             std::vector<double> splineResult = spline.part(i).polynom.getCoefs();
 
             // Check contraints for first endpoint and then the second endpoint in the polynomial
-            for (int k = 0; k < 2; k++) {
+            for (size_t k = 0; k < 2; k++) {
```

Comment:
Cast things to int rather than making things size_t, don't let unsigned infect everything that's how you get bugs.
If somewhere for instance you wanted to do k-1 suddenly it's a massive number.
So often you have people go `if (k - 1 < 0)` and that statement is literally never true with unsigned math.

Rather than making everything unsigned to avoid warnings downstream, make the downstream stuff signed